### PR TITLE
Reference Paths_X alternative in README

### LINKED ISSUE #3: Discuss alternative(s) in README
Maybe the README could discuss alternatives.

E.g. I use the version provided by the `cabal`-generated `Paths_XXX` package:
```haskell
module Version ( version ) where

import Data.List ( intercalate )
import Data.Version ( Version(versionBranch) )

import qualified Paths_PACKAGE as Paths

-- | The program version obtained from the cabal file.

version :: String
version = intercalate "." $ map show $ versionBranch Paths.version
```
Of course, this requires that `cabal build` has run, but maybe this runs anyway at compile time.

### PR DESCRIPTION
**Pull request by tbidne**
Resolves #3.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,3 +16,31 @@
...
+
+```cabal
+cabal-version:      2.4
+name:               foo
+version:            0.1
+
+...
+
+library
+  other-modules: Paths_foo
```
Comment (andreasabel) [inline README.md]: cabal now additionally wants:
```
  autogen-modules: Paths_foo
```

File: README.md [hunk 2]
```diff
@@ -16,3 +16,31 @@
...
+```haskell
+import Data.List (intercalate)
+import Data.Version (Version(versionBranch))
+import Paths_foo qualified as Paths
+
+vsn :: String
+vsn = intercalate "." $ fmap show $ versionBranch Paths.version
+```
+
+See the [source](https://github.com/haskell/cabal/blob/master/Cabal/src/Distribution/Simple/Build/PathsModule/Z.hs) regarding the API.
```
Comment (andreasabel) [inline README.md]: There is also the docs at https://cabal.readthedocs.io/en/3.10/cabal-package.html#accessing-data-files-from-package-code

**Review by andreasabel — COMMENTED** [review]
Looks good, some suggestions for improvement.

File: README.md (hunk 2 above)
Comment (tbidne) [reply README.md]: Oh that is much better. It appears my quarrel is with the search function, as that did not show up in the hits for "paths_", presumably due to the backticks.

**Review by andreasabel — APPROVED** [review]

**Pull request merged**